static/app/views/onboarding/components/useOnboardingQueryParams.tsx

```diff
@@ -0,0 +1,73 @@
+import {useCallback, useMemo} from 'react';
+
+import type {ProductSolution} from 'sentry/components/onboarding/productSelection';
+import {decodeList} from 'sentry/utils/queryString';
+import {useLocation} from 'sentry/utils/useLocation';
+import {useNavigate} from 'sentry/utils/useNavigate';
+
+type QueryValues = {
+  /**
+   * Used to show product selection for certain platforms, e.g. javascript
+   */
+  product?: ProductSolution[];
+  /**
+   * Used to show the loader script for when the platform is javascript
+   */
+  showLoader?: boolean;
+  /**
+   * Used to show or not the integration onboarding for certain platforms, e.g. AWS (python)
+   */
+  showManualSetup?: boolean;
+};
+
+const ONBOARDING_QUERY_PARAMS: (keyof QueryValues)[] = [
+  'showManualSetup',
+  'showLoader',
+  'product',
+];
+
+export function useOnboardingQueryParams(): [
+  params: Partial<QueryValues>,
+  setParams: (newValues: Partial<QueryValues>) => void,
+] {
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  const params = useMemo(() => {
+    const values = ONBOARDING_QUERY_PARAMS.reduce((acc, key) => {
+      const value = location.query[key];
+      acc[key] =
+        value === 'true'
+          ? true
+          : value === 'false'
+            ? false
+            : Array.isArray(value)
+              ? decodeList(value)
+              : value;
```

Comment:
I couldn't find it, but maybe there is a function to handle this

Reply:
IMHO this is hard to read, if there is no function to handle this, this would be good use case to add one, at least in this file, and we would avoid the use of triple nested ternary operator 😅

Reply:
yes, I don't like this either and was hoping someone could point me to the magical function.

Reply:
You will probably want to use `useLocationQuery` here.
Also I would just hardcode reading the 3 params. No need to make it more complicated by iterating over them.

Reply:
thanks @ArthurKnaus! I can do that, but considering that we may have more parameters in the future, I would prefer to keep the logic as it is. The list isn't too long at the moment

Reply:
ok but by using `useLocationQuery` it is better as you suggested :)